code/lightna/magento-frontend/js/common/BodyClickHandler.js

```diff
@@ -0,0 +1,26 @@
+export class BodyClickHandler {
+    static actions = {}
+
+    constructor() {
+        document.addEventListener('click', (event) => this.handleClick(event));
+    }
+
+    static addActions(actions) {
+        this.actions = { ...this.actions, ...actions }
+    }
+
+    handleClick(event) {
+        const actionElement = event.target.closest('[data-action]');
+        if (actionElement) {
+            const actionType = actionElement.getAttribute('data-action');
+            const actionHandler = BodyClickHandler.actions[actionType];
+            if (actionHandler) {
+                try {
+                    actionHandler(event, actionElement);
+                } catch (error) {
+                    console.error(`Error handling click action: ${action}`, error);
```

Comment:
```suggestion
                    console.error(`Error handling click action: ${action}.`, error);
```

Btw, i'm not sure we need try catch here but lets keep it. It definitely makes sense if many handlers could be fires by single click so that next will run even if first failed, but seems it's not the case, we just wrap standard error to add one more line to the console